app/assets/javascripts/admin/addon/controllers/admin-user-index.js

```diff
@@ -141,31 +145,71 @@ export default Controller.extend(CanCheckEmails, {
 
   actions: {
     impersonate() {
-      return this.model.impersonate();
+      return this.model
+        .impersonate()
+        .then(() => (document.location = getURL("/")))
+        .catch((e) => {
+          if (e.status === 404) {
+            bootbox.alert(I18n.t("admin.impersonate.not_found"));
+          } else {
+            bootbox.alert(I18n.t("admin.impersonate.invalid"));
+          }
+        });
     },
     logOut() {
-      return this.model.logOut();
+      return this.model
+        .logOut()
+        .then(() => bootbox.alert(I18n.t("admin.user.logged_out")));
     },
     resetBounceScore() {
       return this.model.resetBounceScore();
     },
     approve() {
       return this.model.approve(this.currentUser);
     },
+
+    _formatError(event) {
+      return `http: ${event.status} - ${event.body}`;
+    },
+
     deactivate() {
-      return this.model.deactivate();
+      return this.model
+        .deactivate()
+        .then(() => window.location.reload())
```

Comment:
Can we not use Ember's routing/reload here? I don't think we need to reload the whole app when the model changes. (I realize for impersonate we have to reload.)